core/src/main/java/feign/ResponseInterceptor.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2012-2022 The Feign Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package feign;
+
+/**
+ * Zero or more {@code ResponseInterceptor} may be configured for purposes such as verify or modify
+ * headers of response, verify the business status of decoded object. No guarantees are given with
+ * regards to the order that interceptors are applied. Once interceptors are applied,
+ * {@link ResponseInterceptor#beforeDecode(Response)} is called before decode method called,
+ * {@link ResponseInterceptor#afterDecode(Object)} is called after decode method called.
+ */
+public interface ResponseInterceptor {
+
+  /**
+   * Called for response before decode, add data on the supplied {@link Response} or doing
+   * customized logic
+   *
+   * @param response
+   * @return
+   */
+  void beforeDecode(Response response);
```

Comment:
thinking if we could/should have a single method only.... may advantage is could be a functional interface.  Also, if someone needs to share context between before and after it would be much easier.

We would have something similar to http filters or around aspects... one argument would have a `proceed` method that  run feign internal code

Reply:
If make a proceed argument like around aspect, the user need to call the proceed explicitly. It seems like a little confused for user. 
May combine the before and after method to one method with two arguments (response and object) and invoke it only after decode, that could be easier to use

Reply:
I think that is fine, since you are intercepting the call, you need to tell it when to run.

IMHO, that offers much better coding possibilities for anyone using it.  Like how to share content from before and after methods.... 

Aspects work like that, filters work like that, I think we will be fine as long is documented.